partcad/src/partcad/assembly.py

```
@@ -45,12 +40,11 @@ def __init__(self, config={}):
 
     async def do_instantiate(self):
-        async with self.locked():
+        if len(self.children) == 0:
+            self._wrapped = None  # Invalidate if any
+            await pc_thread.run(self.instantiate, self)
             if len(self.children) == 0:
-                self.shape = None  # Invalidate if any
-                await pc_thread.run(self.instantiate, self)
-                if len(self.children) == 0:
-                    pc_logging.warning("The assembly %s:%s is empty" % (self.project_name, self.name))
+                pc_logging.warning("The assembly %s:%s is empty" % (self.project_name, self.name))
```

Ruff (0.8.2)

Use format specifiers instead of percent format

Replace with format specifiers

(UP031)